pkg/tests/cross-tests/input_cross_test.go

```diff
@@ -217,3 +217,203 @@ func TestInputsEmptyConfigModeAttrSet(t *testing.T) {
 		SkipCompareRawConfig: true,
 	})
 }
+
+func TestOptionalSetNotSpecified(t *testing.T) {
+	skipUnlessLinux(t)
+	runCreateInputCheck(t, inputTestCase{
+		Resource: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"f0": {
+					Optional: true,
+					Type:     schema.TypeSet,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"x": {Optional: true, Type: schema.TypeString},
+						},
+					},
+				},
+			},
+		},
+		Config:               tftypes.NewValue(tftypes.Object{}, map[string]tftypes.Value{}),
+		SkipCompareRawConfig: true,
+	})
+}
+
+func TestInputsEmptyCollections(t *testing.T) {
+	skipUnlessLinux(t)
+	// TODO[pulumi/pulumi-terraform-bridge#1971]
+	t.Skipf("We don't handle missing collections correctly")
+	config := tftypes.NewValue(tftypes.Object{}, map[string]tftypes.Value{})
+
+	// signifies a block
+	resourceElem := &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"x": {Optional: true, Type: schema.TypeString},
+		},
+	}
+
+	// signifies an attribute
+	schemaElem := &schema.Schema{
+		Type: schema.TypeMap,
+	}
+
+	for _, tc := range []struct {
+		name       string
+		maxItems   int
+		typ        schema.ValueType
+		elem       any
+		configMode schema.SchemaConfigMode
+	}{
+		{"list block", 0, schema.TypeList, resourceElem, schema.SchemaConfigModeAuto},
+		{"set block", 0, schema.TypeSet, resourceElem, schema.SchemaConfigModeAuto},
+		// This isn't quite valid but should work
```

Comment:
What does it mean isn't quite valid? Does it pass schema validators?

Reply:
Maps can't have a Resource Elem - it get's casted to a map of string instead https://github.com/hashicorp/terraform-plugin-sdk/blob/1f499688ebd9420768f501d4ed622a51b2135ced/helper/schema/core_schema.go#L79

it becomes an attribute anyway after.